code/modules/integrated_electronics/subtypes/reagents.dm

```diff
@@ -69,6 +69,34 @@
 			set_pin_data(IC_OUTPUT, 2, WEAKREF(src))
 			push_data()
 
+// Hydroponics trays have no reagents holder and handle reagents in their own snowflakey way.
+// This is a dirty hack to make injecting reagents into them work.
+// TODO: refactor that.
+/obj/item/integrated_circuit/reagent/proc/inject_tray(obj/machinery/hydroponics/tray, atom/movable/source, amount)
```

Comment:
I get that it's out of scope for your PR, but still, you are taking selsa's dirty hack and make it into a dirtier hack. Ew.

Reply:
I'm not pleased with it, but the hydroponics tray attackby() proc is so messy, it would take me a good while to whittle it down into something that I can use regardless of user.

Reply:
I very highly recommend going over the hydroponics tray code in the future. It's not in the scope of this PR but that code needs weeding.